### PR TITLE
Merge extended

### PR DESCRIPTION
**Pull request by bjoo**
This takes over from PR #13. It is an intiial merge of `extended` and `master`. Review and testing required.

### CODE REVIEW

File: lib/qio/QIO_host_file_conversion.c [hunk 1]
```diff
@@ -1081,7 +1087,7 @@ int QIO_part_to_single( const char filename[], int ildgstyle,
 	  /* Copy LIME type */
 	  lime_type_out = (char *)malloc(strlen(lime_type_in)+1);
 	  if( lime_type_out == NULL ) return QIO_BAD_ARG;
-	  sprintf(lime_type_out,"%s",lime_type_in);
+	  strcpy(lime_type_out,lime_type_in);
```
Comment (maddyscientist) [inline lib/qio/QIO_host_file_conversion.c:1090]: random aside comment: are there other places in QIO that could have `sprintf` -> `strcpy` applied?  `sprintf` is notoriously slow compared to `strcpy`.

**Comment by bjoo** [comment]
There were 1 or 2. I hopefully resolved them in favour of strcpy....

**Review by maddyscientist — APPROVED** [review]
This looks good to me.  Tested with QUDA + QMP + QIO, and with a variety of compilers.  I did find a few warnings with the NVHPC compiler, and a build failure (lack of support for the `__rdtsc` intrinsic).  So I have pushed a small patch to address these issues.

**Comment by bjoo** [comment]
Thank you for the review, tests and fixes @maddyscientist

**Review by mathiaswagner — APPROVED** [review]
QUDA tests passed and I am happy to reduce clutter.

**Review by jcosborn — APPROVED** [review]
I'm getting some warnings in the examples, but that can be fixed after this.

**Comment by mathiaswagner** [comment]
If they are easy to fix better do it now. Would appreciate that for a warning-free Quda build as well.

**Pull request merged**